GameEngine/Agents/GreedyBestFirstAgent.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+
+namespace GameEngine.Agents
+{
+    public class GreedyBestFirstAgent : IAgent
+    {
+        public Queue<SearchNode> solution;
+
+        public Play FormulatePlay(GameState state)
+        {
+            solution = solution ?? GreedyBestFirstSearch(state);
+            var nextNode = solution.Dequeue();
+            if (nextNode.Parent.State != state)
+            {
+                throw new StateMismatchException();
+            }
+            return nextNode.Action;
+        }
+
+        private Queue<SearchNode> GreedyBestFirstSearch(GameState state)
+        {
+            var frontierNodes = new List<SearchNode> { new RootNode(state) };
+            while (true)
+            {
+                if (frontierNodes.Count == 0)
+                {
+                    throw new NoSolutionFoundException();
+                }
+                var node = frontierNodes.RemoveMin(H);
```

Comment:
❓ It was my understanding the "min" should be PathCost + H?

Reply:
If we added that tie-breaking behavior with minimal refactoring, it would live in `RemoveMin`. More evidence that it should probably be abstracted away. I think we'll cover this topic while talking about [this other comment ](https://github.com/rpherbig/hoot-owl-hoot/pull/8#discussion_r346402060), so I'm resolving this thread.